libcaf_core/caf/detail/json.cpp

```diff
@@ -136,9 +207,31 @@ void assign_value(Escaper escaper, Consumer& consumer, Iterator first,
   consumer.value(str);
 }
 
-} // namespace
+// Check 4 hex digits follow the '\u' escape sequence.
+template <class ParserState, class Action>
+void read_4_hex(ParserState& ps, Action action) {
```

Comment:
Shouldn't just emit single characters but the code point. Then, `read_code_point` might also be a better name. Further, the second parameter is a `Consumer`, not an action.

Reply:
Refactored.